frame/nfts/src/lib.rs

```diff
@@ -939,10 +1016,12 @@ pub mod pallet {
 				ensure!(permitted, Error::<T, I>::NoPermission);
 			}
 
-			details.approved = Some(delegate);
+			details
+				.approvals
+				.try_insert(delegate.clone(), deadline)
```

Comment:
It would probably make sense to set the deadline to `current_block + deadline`, will change this.